test/unit/managers/ConfigManager.ts

```diff
@@ -157,3 +170,125 @@ test('should have enableOnSession flag', t => {
 
   t.not(fakeMergedConfig.enableOnSession, undefined);
 });
+
+/**
+ * Check Restricted Origin Tests
+ */
+test('should match origins', async t => {
+  const checkRestrictedOriginSpy = sinonSandbox.spy(ConfigHelper, "checkRestrictedOrigin");
+  const doesOriginMatchSpy = sinonSandbox.spy(ConfigHelper, "doesCurrentOriginMatchConfigOrigin");
+  sinonSandbox.stub(OneSignal, "delayedInit");
+  try{
+    await initOneSignalOnUrlWithConfigOrigin("https://example.com", "https://example.com");
+  } catch (e) {
+    // pass
+  }
+
+  t.true(doesOriginMatchSpy.calledOnce);
+  t.true(doesOriginMatchSpy.returnValues[0]);
+  t.false(checkRestrictedOriginSpy.threw());
+});
+
+test('should match origins with www. variant in config', async t => {
+  const checkRestrictedOriginSpy = sinonSandbox.spy(ConfigHelper, "checkRestrictedOrigin");
+  const doesOriginMatchSpy = sinonSandbox.spy(ConfigHelper, "doesCurrentOriginMatchConfigOrigin");
+  sinonSandbox.stub(OneSignal, "delayedInit");
+  try{
+    await initOneSignalOnUrlWithConfigOrigin("https://example.com", "https://www.example.com");
+  } catch (e) {
+    // pass
+  }
+
+  t.true(doesOriginMatchSpy.calledOnce);
+  t.true(doesOriginMatchSpy.returnValues[0]);
+  t.false(checkRestrictedOriginSpy.threw());
+});
+
+test('should match origins with www. variant on page', async t => {
```

Comment:
Seems www. and non-www are different origins according to this.
https://developer.mozilla.org/en-US/docs/Glossary/origin

HTTPS needs be strict if "My site is fully HTTPS" is used without a subdomain.